### PR TITLE
Added capability to use (and enforce) native netty transport in azure-cosmos-spark

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
This PR adds a capability to use (and enforce) native netty transport in azure-cosmos-spark. The native transport is more efficient - esepcially when the number of TCP connections being used is high.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md [hunk 1]
```diff
@@ -24,6 +24,7 @@
...
+| `spark.cosmos.enforceNativeTransport`                             | `false` | Indicates whether native netty transport should be enforced. If true, on clinet creation an exception will be trown if netty is not able to use native transport. The native transport is more efficient especiall when there is a high number of connections.                                                                                                                                                                                                                                                                                                                                                |
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md:27]: wonder when this config will be needed

if the native library exists, then why customer would want to turn it off.  And if they want to enforce it and the native library does not exists, then it means we have a bug in our library as shading?

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md [hunk 2]
```diff
@@ -24,6 +24,7 @@
...
+| `spark.cosmos.enforceNativeTransport`                             | `false` | Indicates whether native netty transport should be enforced. If true, on clinet creation an exception will be thrown if netty is not able to use native transport. The native transport is more efficient especially when there is a high number of connections.                                                                                                                                                                                                                                                                                                                                                |
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md:27]:
```suggestion
| `spark.cosmos.enforceNativeTransport`                             | `false` | Indicates whether native netty transport should be enforced. If true, on client creation an exception will be thrown if netty is not able to use native transport. The native transport is more efficient especially when there is a high number of connections.                                                                                                                                                                                                                                                                                                                                                |
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java [hunk 3]
```diff
@@ -1218,7 +1218,7 @@ private void ifThrowIllegalArgException(boolean value, String error) {
     private void logStartupInfo(StopWatch stopwatch, CosmosAsyncClient client) {
         stopwatch.stop();
 
-        if (logger.isInfoEnabled()) {
+        if (logger.isWarnEnabled()) {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java:1221]: thanks for fixing this :)

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**